### ISSUE TITLE
add flux_whoami()

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#1544]
We identified a use case where the "none" signing mechanism would be OK to use for job submission:  when the submitting user is the same as the instance owner.  This can work because the instance owner can use regular connector authentication to authenticate the user, and the IMP can allow the instance owner to launch processes as itself (real userid == the job userid).

This is a nice thing because probably it will be the norm for instances other than the system instance.  In addition, high througput workloads running in a dedicated instance as a single user will benefit from lower job submission overhead/latency.

However there is no easy way for a `flux run` command to determine if it's running as the instance owner.

Let's add an API call that allows a user to fetch its userid and rolemask, e.g.
```c
flux_future_t *flux_whoami (flux_t *h);
int flux_whoami_get_userid (flux_future_t *f, uint32_t *userid);
int flux_whoami_get_rolemask (flux_future_t *f, uint32_t *rolemask);
```

`flux run` can then call
```c
if (!(f = flux_whoami (h)))
    fatal();
if (flux_whoami_get_rolemask (f, &rolemask) < 0)
    fatal();
if ((rolemask & FLUX_ROLE_OWNER))
    mech = "none";
else
    mech = NULL;
# flux_sign_wrap() modified to accept "mech" parameter, NULL to take default.
```

**Comment by grondo** [comment]
That sounds good!

**Comment by grondo** [comment]
I think this solution was obviated by the `flux::owner` getopt implemented in 1a2d9dd5fa9dce7b157b926ca61dcb5d37d06a14
